File: app/src/main/java/life/plank/juna/zone/data/network/model/League.java
package life.plank.juna.zone.data.network.model;

import android.support.annotation.ColorRes;

import lombok.Data;

@Data
public class League {
    private Integer id;
    private Long foreignId;
    private String name;
    private Boolean isCup;
    private String seasonName;
    private String countryName;
    private String thumbUrl;
    private int leagueLogo;

    @ColorRes
    private int dominantColor;

    public League(String name, Boolean isCup, String seasonName, String countryName, String thumbUrl, int dominantColor, int leagueLogo) {
        this.name = name;
        this.isCup = isCup;
        this.seasonName = seasonName;
        this.countryName = countryName;
        this.thumbUrl = thumbUrl;
        this.dominantColor = dominantColor;
        this.leagueLogo = leagueLogo;
    }
}

File: app/src/main/java/life/plank/juna/zone/data/model/League.kt
package life.plank.juna.zone.data.model

import android.os.Parcelable
import androidx.annotation.DrawableRes
import androidx.room.Entity
import androidx.room.Ignore
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Entity
@Parcelize
data class League @Ignore constructor(
        var name: String = "",
        var isCup: Boolean,
        var seasonName: String?,
        var countryName: String?,
        var thumbUrl: String?,
        var dominantColor: Int?,
        @DrawableRes var leagueLogo: Int,
        @SerializedName("foreignId") @Expose var id: Long = 0
) : Parcelable {
    @Ignore constructor(id: Long, name: String, isCup: Boolean, seasonName: String?, countryName: String?, thumbUrl: String?, dominantColor: Int?, leagueLogo: Int) :
            this(name, isCup, seasonName, countryName, thumbUrl, dominantColor, leagueLogo, id)

    constructor() : this("", false, "", "", "", 0, 0, 0)
}